src/nstack.c

```diff
@@ -326,30 +326,59 @@ void nstack_stop(void)
 
 int main(int argc, char *argv[])
 {
+    /* getopt support */
+    int opt; 
+    char *intf, *mac;
+    while((opt = getopt(argc, argv, "i:m:")) != -1){
+        switch (opt) {
+            case 'i':
+                // interface, ex: veth0, eth0 ... 
+                // intf = malloc(sizeof(char)*strlen(optarg)+1);
+                intf = optarg;
+                break;
+            case 'm':
+                // mac Address
+                mac = optarg;
+                break;
+            default: /* not match */
+                fprintf(stderr, "Usage: %s [-i <interface>] [-m <mac_addr>] [-h]\n",argv[0]);
+                exit(EXIT_FAILURE);
+        }
+    }
+    // argv[1] = Network Interface, ex: eth0, veth1, ... etc.
```

Comment:
You shall not use "ex" here. Instead, the abbreviation e.g. stands for Latin exempli gratia, which means "for example. "